### ISSUE TITLE
Auto provision namespaces on tenant creation 

### ISSUE THREAD

**Issue by nmarcetic** [clastix/capsule#325]
# Describe the feature
It would be handy if the tenant namespace can be auto-provisioned on tenant creation. When Bob, as a cluster-admin provisions new tenant oil with three namespaces development, production and test, they are auto-provisioned (e.g namespace admin Alis don't need to create them). It's a bit different from k8s native expi, but very handy in cases when "container as a service" expi is not a case. 
 
# What would the new user story look like?
Cluster admin creates a new tenant with `namespaceAutoProvision` flag 
```
apiVersion: capsule.clastix.io/v1alpha1
kind: Tenant
metadata:
  name: oil
spec:
  owner:
    name: alice
    kind: User
  namespaceQuota: 3
  namespaceAutoProvision: true
status:
  namespaces:
    oil-development
    oil-production
    oil-test
```

# Expected behavior
When tenant admin Alise lists namespaces she gets 
```
  oil-development
  oil-production
  oil-test
```
And she is set as a namespace admin. 

I am not sure if this is already possible? I didn't saw such [use case](https://github.com/clastix/capsule/tree/master/docs/operator/use-cases) , but if such option exists , please reference it. Thanks

**Comment by prometherion** [comment]
Hey @nmarcetic, thanks for the feature report!

It seems we already discussed something similar on https://github.com/clastix/capsule/issues/174#issuecomment-749757971, although different.

> tl;dr; set-up a post-install Helm hook to create the Namespaces upon Tenant creation. Obviously, this works just with Helm and/or any other GitOps solution that provides these kinds of hooks.

I don't think we want to let Capsule create on behalf of the Tenant owner a set of desired Namespace resources, since we embraced the self-service approach since the beginning of the project, but let's open a discussion if you find it's viable.

**Comment by nmarcetic** [comment]
@prometherion Thanks! Basicaly it's the same/duplicated, just not related to the helm. Helm solution is a bit complicated and limited to the helm, but I suppose it can work for me too, thanks to @d-m

Having this In capsule core would be valuable, yes, but I agree about the self-service approach since it's not really k8s native style. I was looking it more like a resource limitation (like node selector or quotes, it's exactly what I want to limit tenant) to tenant from cluster-admin than doing it on behalf of the tenant (like automating).

**Comment by bsctl** [comment]
@nmarcetic thanks for interest in Capsule! When designed it, we assumed that is not responsibility of Capsule to create namespaces on behalf of the tenant owner. Namespaces self service provisioning is one of the key features in Capsule.

IMO, if you need for namespaces creation in automated way, it can be achieved as part of tenant on boarding procedure, for example.

**Comment by prometherion** [comment]
I couldn't agree more, @bsctl.

> I was looking at it more like a resource limitation (like node selector or quotes, it's exactly what I want to limit tenant) to tenant from cluster-admin than doing it on behalf of the tenant (like automating).

I guess if you play with Namespace quota and automation during the CD phase, you could nail it.

Going to close the issue, feel free to open it back at your wish!

**Comment by nmarcetic** [comment]
Got it. Provisioning In the onboarding stage makes sense, I will go that way, doing it as abstraction and keep capsule handle isolation only. 

Thanks, @prometherion @bsctl kudos for great work!
